### PR TITLE
Local Authority user report spreadsheet export

### PR DESCRIPTION
**Pull request by pjshumphreys**
http://trello.com/c/UsiGWFJU

### CODE REVIEW

File: server/routes/reports/localAuthorityReport/template/xl/sharedStrings.xml [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+        <t>Local authority name</t>
+    </si>
+    <si>
+        <t>Local Authority</t>
+    </si>
+    <si>
+        <t>Parent name</t>
+    </si>
+    <si>
+        <t></t>
```
Comment (wozzer72) [inline server/routes/reports/localAuthorityReport/template/xl/sharedStrings.xml]: What does an empty element mean? There are more than one empty elements in this XML structure.

File: server/routes/reports/localAuthorityReport/user.js [hunk 2]
```diff
@@ -1,207 +1,1400 @@
...
+            );
+        } break;
+
+        case 'H': {
+          redIifMissing(
+              putString,
+              cellToChange,
+              reportData.workers[row].mainJob,
+              columnText,
+
```
Comment (wozzer72) [inline server/routes/reports/localAuthorityReport/user.js]: Are you saving a file here? If so, then best check with Nasir on the container file storage capability - and to estimate (or better still) calculate the expected largest file size to ensure use of local file system can be supported?

And you'll need to cope with multiple concurrent requests. How many concurrent requests, I don't know.

Are you deleting the file - to maintain storage availability?

File: server/routes/reports/localAuthorityReport/template/xl/sharedStrings.xml (hunk 1 above)
Comment (pjshumphreys) [reply server/routes/reports/localAuthorityReport/template/xl/sharedStrings.xml]: The shared strings as used via an index, and this template was creted from a spreadsheet with sample data. In order for the Ids not to change but not share the sample data I've just blanked them out

File: server/routes/reports/localAuthorityReport/user.js (hunk 2 above)
Comment (pjshumphreys) [reply server/routes/reports/localAuthorityReport/user.js]: No files are being saved. They are purely created in memory then streamed to the client then garbage collected

Comment (wozzer72) [reply server/routes/reports/localAuthorityReport/user.js]: Excellent

**Pull request merged**